pycsw/ogc/gml/gml3.py

```diff
@@ -47,6 +47,11 @@ def _poslist2wkt(poslist, axisorder):
     tmp = poslist.split()
     poslist2 = []
 
+    if len(tmp) < 4:
```

Comment:
Shouldn't this be `<5`, i.e. only a triangle with the starting point twice is allowed having the least number of coordinates?

Reply:
A triangle in WKT would look like:

```POLYGON((30 10, 40 40, 10 20, 30 10))```

Reply:
Yes, that's clear, but I am still confused why the check uses 4, this would mean `48.1537 16.2825 49.1537` is not valid, but  `48.1537 16.2825 49.1537 17.2825` is valid, which is theoretically a line, not a polygon. In my opinion `30 10 40 40 10 20 30 10` is a valid polygon, but `30 10 40 40 10 20` not, so the check should be `<8` (sorry the '<5' in the comment above is wrong.)

Reply:
The `_poslist2wkt` function works against a posList in the context of either a line or point so putting 4 would minimally satisfy the line case (2 direct positions for line, 4 direct positions for polygon).  Beyond a validating parser, there's no additional checks done (we were careful not to go to deep into GML, which would be overkill IMHO for the pycsw use case), so the expectation would be for the underlying data store to throw an exception on a bad geometry.

Having said this, I've added a fix to catch this nonetheless depending on the line or polygon geometry type.

Reply:
Okay, thank you for the clarification!